### DISCUSSION TITLE
No Filesystem on STM32WL Nucleo board

### DISCUSSION THREAD

**Discussion by DuaneKaufman** [micropython/micropython#12983] [STM32 / Pyboard] [answered]
Dear Sirs,
I purchased an STM32WL Nucleo board, and installed Micropython (both stable and preview - same outcome), but there is no filesystem. I can access the REPL fine:

`duane@HP-Spectre:~$ mpremote repl
Connected to MicroPython at /dev/ttyACM0
Use Ctrl-] or Ctrl-x to exit this shell
roPython v1.22.0-preview.147.ga00c9d56d on 2023-11-13; NUCLEO-WL55 with STM32WL55JCI7
Type "help()" for more information.
>>> import os
>>> os.chdir("/")
>>> os.listdir(".")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 19] ENODEV
>>> 
`
What am I doing wrong?

Sincerely,
Duane

**Comment by davefes** [comment]
Does the Nucelo get mounted on the host?

Just tried your commands on a STM32F411 BlackPill and same result.

Change to `os.listdir()`

**Reply by DuaneKaufman** [reply]
Nucleo was being mounted by host.
I needed to create a littlefs on the Nucleo, and I can now store files there.

**Comment by projectgus** [comment, marked as answer]
The STM32WL55 is pretty constrained on flash space, so it doesn't create any filesystem by default.

You can still create one in the available space, take a look at this comment here which has the steps to do that:
https://github.com/orgs/micropython/discussions/11708#discussioncomment-6118083

Note that the filesystem will be pretty small (in the linked post it was 24KB, in the latest master branch versions we "freeze" the lora driver into the STM32WL55 build so the available space for the filesystem will be even less - but on the plus side you now don't have to find a place to put the LoRa driver!)

Suggest pre-compiling any sizeable Python files with [mpy-cross](https://docs.micropython.org/en/latest/develop/gettingstarted.html#building-the-micropython-cross-compiler) and then copy the .mpy file instead of the .py file to the filesystem. This can save a lot of file size.

If it turns out that there's still not enough space for all the code you need to include, a possible workaround is to build a [custom firmware that "freezes" in the code](https://docs.micropython.org/en/latest/reference/manifest.html).

**Reply by DuaneKaufman** [reply]
Thanks! Works for me.
